crates/ruff_linter/src/settings/mod.rs

```diff
@@ -206,10 +390,8 @@ impl Default for LinterSettings {
 }
 
 /// Given a list of patterns, create a `GlobSet`.
-pub fn resolve_per_file_ignores(
-    per_file_ignores: Vec<PerFileIgnore>,
-) -> Result<Vec<(GlobMatcher, GlobMatcher, RuleSet)>> {
-    per_file_ignores
+pub fn resolve_per_file_ignores(per_file_ignores: Vec<PerFileIgnore>) -> Result<PerFileIgnores> {
```

Comment:
Perhaps makes sense as an associated method on `PerFileIgnores`, now that the struct exists? But not blocking feedback.